draft-ietf-quic-tls.md

```diff
@@ -1059,6 +1059,27 @@ could either be absent or it could indicate that an address validation token is
 not present.
 
 
+### Stateless Address Validation
+
+A server can use the cookie extension to store all state necessary to continue
+the connection.  This allows a server to avoid committing state for clients that
+have unvalidated source addresses.
+
+For instance, a server could use a statically-configured key to encrypt the
+information that it requires and include that information in the cookie.  In
+addition to address validation information, a server that uses encryption also
+needs to be able recover the hash of the ClientHello and its length, plus any
+information it needs in order to reconstruct the HelloRetryRequest.
+
+A client proceeds as though the server were stateful.  It constructs a second
+ClientHello and sends it following the initial ClientHello.  Note that this
+means a server that doesn't maintain state between the intial and second
+ClientHello will receive the second ClientHello in STREAM frames with a non-zero
+offset.  The resulting gap in the received stream data needs to be ignored and
```

Comment:
Here is a proposal that might help simplify handshake without interfering with 0-RTT stream semantics. For 1-rtt it could be required that streams are not permitted before handshake is complete. https://github.com/quicwg/base-drafts/issues/489